internal/store/block/raft/appender.go

```diff
@@ -572,7 +569,7 @@ func (a *appender) doWakeup(ctx context.Context, commit int64) {
 		if waiter.offset > commit {
 			break
 		}
-		close(waiter.c)
+		a.waiterC <- waiter
```

Comment:
CommitWaiter is not minimal interface to expose.

Reply:
segment package used CommitWaiter